cips/discourage-memo-usage.md

```diff
@@ -0,0 +1,139 @@
+author: Rootul Patel (@rootulp), Gabriel Blaut (@nashqueue)
+discussions-to: https://forum.celestia.org/t/cip-discourage-memo-usage/1508
+status: Draft
+type: Standards Track
+category: Core
+created: 2024-01-21
+---
+
+## Abstract
+
+This proposal aims to discourage the use of transaction memos in Celestia by modifying two governance parameters:
+
+- Decrease `auth.MaxMemoCharacters` from 256 to 16.
+- Increase `auth.TxSizeCostPerByte` from 10 to 16.
+
+## Motivation
+
+Transactions on Celestia may optionally include a user-specified memo. The Cosmos SDK describes the memo as "a note or comment to send with the transaction". The memo field is often used by centralized exchanges to uniquely identify the user depositing into an exchange. The memo field has also been used by IBC relayers to tag IBC transactions with the relayer's name and software version.
+
+Recently, a number of [inscriptions](https://trustmachines.co/glossary/inscriptions/) projects on Cosmos chains have built entire protocols based on the information transmitted via the memo field. One such project that launched on Celestia is CIAS. For example, see [this tx](https://celenium.io/tx/0facb4c1d38e12574ccef8ed2a22d99300b6c91f08470cd75000d30f9f45412e) which has a memo field that base64 decodes to
+
+```txt
+data:,{"op":"cancel","amt":3230000,"tick":"cias","p":"cia-20"}
+```
+
+based on the [CIAS docs](https://docs.cias.wtf/) this memo cancels an inscription listing. There are similar memos to deploy, mint, transfer, and list inscriptions.
+
+On one hand, it is exciting to see protocols launch on top of Celestia. On the other hand, this usage poses risks to the Celestia network.
+
+Celestia was designed to support the publication of arbitrary data but the memo field is not the ideal mechanism to publish such data. One of the design goals stated in the original [LazyLedger](https://arxiv.org/pdf/1905.09274.pdf) paper is:
+
+> Application message retrieval partitioning. Client nodes must be able to download all of the messages relevant to the applications they use from storage nodes, without needing to downloading any messages for other applications.
+
+For this reason, blob data is scoped to an application-specific namespace. However, the transaction memo field is not scoped to an application-specific namespace. Instead, it pollutes the reserved [`TRANSACTION_NAMESPACE`](https://celestiaorg.github.io/celestia-app/specs/namespace.html#reserved-namespaces). This is undesirable because partial nodes that want to verify the current Celestia state must download and execute all transactions in the `TRANSACTION_NAMESPACE` (including memos).
+
+As of January 17, 2024, 0.51 GiB of data has been published to the Celestia network via memos and 1.46 GiB of data has been published via blob data.
+
+This proposal seeks to realign incentives so that protocol builders are encouraged to favor application-specific namespace blob data over memo fields.
+
```

Comment:
- [ ] Add specification section with current values and values that will be changed to after this CIP

Reply:
Good idea resolved via 99bdcb93558544fc205a0ff81e7d84ec3890416a